File: packages/ui/db/entity/Coordinate.ts
import { type ICoordinateLiteral } from "models";
import { EntitySchema } from "typeorm";

// Embeddable Coordinate columns
export type Coordinate = ICoordinateLiteral;

const CoordinateSchema = new EntitySchema<Coordinate>({
  name: "coordinate",
  columns: {
    lat: {
      type: "decimal",
    },
    lng: {
      type: "decimal",
    },
  },
});

export default CoordinateSchema;

File: go-server/internal/models/coordinate.go
package models

type Coordinate struct {
	Lat float64 `json:"lat" sqlp:"Lat"`
	Lng float64 `json:"lng" sqlp:"Lng"`
}

func (c Coordinate) IsZero() bool {
	return c.Lat == 0 && c.Lng == 0
}
